controllers/statusreport/statusreport_adapter.go

```diff
@@ -86,17 +90,42 @@ func (a *Adapter) EnsureSnapshotTestStatusReportedToGitProvider() (controller.Op
 			return controller.RequeueWithError(err)
 		}
 	}
+	integrationTestScenarios, err := a.loader.GetRequiredIntegrationTestScenariosForApplication(a.client, a.context, a.application)
+	if err != nil {
+		return controller.RequeueWithError(err)
+	}
 
-	if gitops.IsSnapshotMarkedAsPassed(a.snapshot) || gitops.IsSnapshotMarkedAsFailed(a.snapshot) || gitops.IsSnapshotMarkedAsInvalid(a.snapshot) {
-		a.logger.Info("Status has been reported successfully, now in the process of removing the finalizer")
-		// Remove the finalizer from all the Integration PLRs related to the Snapshot
-		err = helpers.RemoveFinalizerFromAllIntegrationPipelineRunsOfSnapshot(a.client, a.logger, a.context, *a.snapshot, helpers.IntegrationPipelineRunFinalizer)
-		if err != nil {
-			return controller.RequeueWithError(err)
-		}
-		a.logger.Info("Successfully removed the finalizer from all the Integration PLRs related to current snapshot", "snapshot.Name", a.snapshot.Name)
+	testStatuses, err := gitops.NewSnapshotIntegrationTestStatusesFromSnapshot(a.snapshot)
+	if err != nil {
+		return controller.RequeueWithError(err)
 	}
 
+	for _, integrationTestScenario := range *integrationTestScenarios {
```

Comment:
we want to unlock all pipeline runs right? Not just required

Reply:
Yes you are correct, this is also fixed by the change to loop over the existing related statuses.